### ISSUE TITLE
DeepFool attack issue 

### ISSUE THREAD

**Issue by a7b23** [bethgelab/foolbox#145]
I was trying deepfool attack on Mnist. For the 10,000 test images, the attack returned around 9500 of them, but not all of them are adversarial as about for 30% of these the models predictions is same as the ground truth label.
So why are the images getting returned even if they are not adversarial?

The deepfool paper reports very good performance of the attack with an adversarial performance of around 95%. However I am getting only around 70%.

**Comment by jonasrauber** [comment]
Most likely you have a preprocessing mismatch between what you tell Foolbox and what you use. Please provide code. Foolbox only returns images that fool the network it attacks.

**Comment by a7b23** [comment]
I have not used any additional preprocessing. Here is the code snippet -


model = keras.models.load_model("model.h5")
(x_train, y_train), (x_test, y_test) = mnist.load_data()

foolbox_model = foolbox.models.KerasModel(model, bounds=(0, 1))

adversarial_images = []
original_labels = []
for i in range(len(x_test)) :
    attack = foolbox.attacks.DeepFoolAttack(foolbox_model)
    img = x_test[i]
    label = y_test[i]
    adversarial = attack(img, label,steps = 100)
    if adversarial is None:
        continue
    else :
        adversarial_images.append(adversarial)
        original_labels.append(label)

adversarial_images = np.array(adversarial_images)
original_labels = np.array(original_labels)

model_predictions = model.predict(adversarial_images)

print('fraction of non adversarial images are ',np.mean(np.argmax(model_predictions,1) == original_labels))

**Comment by jonasrauber** [comment]
Can you try `model_predictions = foolbox_model.batch_predictions(adversarial_images)` instead of `model_predictions = model.predict(adversarial_images)`?

Can you report the printed value in both cases?

**Comment by a7b23** [comment]
Hi, I did the following -
`model_predictions = model.predict(modified_images)`
`foolbox_model_predictions = foolbox_model.batch_predictions(modified_images)`
`print('The fraction of same predictions is',np.mean(np.argmax(model_predictions,1) == np.argmax(foolbox_model_predictions,1)))`

And I was getting wierd behaviour wherby sometimes the output would come as 1(or the predictions are same) whereas sometimes it would come as 0.87(for some examples the predictions differ).

Upon closer inspection I found that sometimes its happening that 2 classes are getting almost the same probability. For example for one test example, the model prediction is - 
```
[1.0301829e-06 4.7053501e-04 4.9774656e-01 3.9901929e-03 9.1776684e-09
 2.3160034e-09 2.1843649e-10 4.9774754e-01 4.2688556e-05 1.4837605e-06]
```
 whereas the foolbox model prediction is - 
```
[ -3.3498933   2.77424     9.738218    4.9119654  -8.07061    -9.447543
 -11.808646    9.738218    0.374301   -2.985051 ]
```
Thus here classes 2 and 7 are getting almost the same values for both the predictions , thereby leading to a difference between their argmax.
